UnityProject/Assets/Scripts/Objects/Closets/ClosetControl.cs

```diff
@@ -483,7 +483,8 @@ public bool WillInteract(HandApply interaction, NetworkSide side)
 	public void ServerPerformInteraction(HandApply interaction)
 	{
 		// Is the player trying to put something in the closet?
-		if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Emag) && Validations.EmagHasCharges(interaction.HandObject))
+		if (Validations.HasItemTrait(interaction.HandObject, CommonTraits.Instance.Emag) 
+			&& interaction.HandObject.GetComponent<Emag>().EmagHasCharges())
```

Comment:
```suggestion
			&& interaction.HandObject.TryGetComponent<Emag>(out var emag) && emag.EmagHasCharges())
```
This will allow you to just reference `emag` for when you call `UseCharge()`, removing a `GetComponent` call.
Similar situation for the other areas where the emag is used.

Reply:
nice! thank you, didn't know that was a thing